### PR TITLE
Big refactoring of everything

### PR DESCRIPTION
**Pull request by AlexWayfer**
* Tune RuboCop
* __Remove `--delay` option__
* __Rename `FileWatcher` to `Filewatcher`__
* Use `require_relative`
* Move Trollop banner to separate file
* Small syntax impromenets
* Create `Filewatcher::Env` and `Filewatcher::Runner` sub-classes
* Move `Filewatcher::VERSION` to separate file
* Remove files for tests (`fixtures`) from git
* Make temp files for test in `tmp` directory insted of `fixtures`
* Clean temp files after tests
* Add more tests
* Improve existing tests
* Reduce code duplicating in tests by `test_helper` with `WatchRun` class
* Move some Filewatcher methods to `private`
* Tune `Rakefile`
* __Replace `sleep` argument for `watch` method to `interval` writer__
* Fix `--exclude` option
* Fix `--immediate` option

@thomasfl, what do you think? Especially about bold list items.

Rename according to the guides, otherwise the files and the gem would have to be called `file_watcher` in downcase.

`--delay` option is useless, especially after #53, and can be realized by `sleep` before command.

Also I want to remove all RuboCop-warnings before merging.

### CODE REVIEW

**Comment by thomasfl** [comment]
Very good Alex! 🥇 

The change of name is a good idea. I didn't know the delay option was useless, so please delete it. 

I do not understand what this is, but I trust your decision:
> Replace sleep argument for watch method to interval writer

**Comment by thomasfl** [comment]
@AlexWayfer I think you can add your name to the Readme file. Maybe describe yourself as the maintainer?

**Comment by AlexWayfer** [comment]
> I didn't know the delay option was useless, so please delete it.

I thought this will help for #46, but later I made fixes and improvements with `changes` system in 57dca878487bfd424826ea81345150b7270b2cab, which prevented multiple execution of the command. So, now `--delay` option makes little sense.

> Replace sleep argument for watch method to interval writer

`Filewatcher.new` has `interval` option, and `Filewatcher#watch` has `sleep` argument for the same purposes. So, I think, this is excessive repetition and delusion, and if you have `Filewatcher` object (instance), and you want to change `interval` value for (next) `watch` call — you can set via `filewatcher.interval = ` now. Example:

Before:

```ruby
filewatcher = Filewatcher.new('**/*', interval: 0.5)
filewatcher.watch { puts 'watch every 0.5 seconds' }
filewatcher.watch(2) { puts 'watch every 2 seconds' }
```

After:

```ruby
filewatcher = Filewatcher.new('**/*', interval: 0.5)
filewatcher.watch { puts 'watch every 0.5 seconds' }
filewatcher.interval = 2
filewatcher.watch { puts 'watch every 2 seconds' }
```

> @AlexWayfer I think you can add your name to the Readme file. Maybe describe yourself as the maintainer?

Okay, thank you :hugs: I'll try to do it neatly.

**Comment by AlexWayfer** [comment]
All RuboCop warnings resolved! But I'm not sure that this way the best way: I took out into module `Filewatcher::Cycles` content of `Filewatcher#watch` method, and split it to separate methods (for pausing stage, watching stage and processing stage).

And I've added myself to `Credits` of `README`, but without `maintener` rank for now :)

Maybe I'll add this after other issues resolving.

**Comment by thomasfl** [comment]
Looking forward to see the finished result. I agree with you suggestion to use this syntax to set interval option:

```
filewatcher.interval = 2
```

File: README.md [hunk 1]
```diff
@@ -326,6 +319,8 @@ This project would not be where it is today without the generous help provided b
 
 *   [Kristoffer Roupé](https://github.com/kitofr): Command line globbing
 
+*   [Alexander Popov](https://github.com/AlexWayfer): Daemon mode, many small fixes and improvements
+
```
Comment (thomasfl) [inline README.md]: Your to modest. ;-)

File: README.md [hunk 2]
```diff
@@ -326,6 +319,8 @@ This project would not be where it is today without the generous help provided b
...
 This gem was initially inspired by [Tom Lieber's blogg posting](http://alltom.com/pages/detecting-file-changes-with-ruby).
```
Comment (thomasfl) [inline README.md]: This blog posting is long gone, but it can be there for reference.

Comment (AlexWayfer) [inline README.md]: I'll add link to [the last live archive version](http://web.archive.org/web/20120208094934/http://alltom.com/pages/detecting-file-changes-with-ruby).

Comment (thomasfl) [inline README.md]: Cool! 😊👍

File: lib/filewatcher/cycles.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class Filewatcher
+  # Module for all cycles in `Filewatcher#watch`
+  module Cycles
+    def main_cycle
```
Comment (AlexWayfer) [inline lib/filewatcher/cycles.rb]: Need to be `private` methods

Comment (AlexWayfer) [inline lib/filewatcher/cycles.rb]: Done.

File: README.md (hunk 2 above)
Comment (AlexWayfer) [inline README.md]: Added.

**Pull request merged**